src/stream.rs

```diff
@@ -159,17 +165,25 @@ impl StreamMap {
 
                 // Enforce stream count limits.
                 match (is_local(id, is_server), is_bidi(id)) {
-                    (true, true) =>
-                        self.peer_max_streams_bidi = self
-                            .peer_max_streams_bidi
-                            .checked_sub(1)
-                            .ok_or(Error::StreamLimit)?,
-
-                    (true, false) =>
-                        self.peer_max_streams_uni = self
-                            .peer_max_streams_uni
-                            .checked_sub(1)
-                            .ok_or(Error::StreamLimit)?,
+                    (true, true) => {
+                        self.local_opened_streams_bidi += 1;
```

Comment:
couldn't this screw up counts if the local peer keeps trying to open streams while the remote peer doesn't update the max? Would it be better to test the condition like `self.local_opened_streams_bidi +1 <= self.peer_max_streams_bidi` and then only increment `self.local_opened_streams_bidi` much later when we actually succeed in `Stream::new()`?

Reply:
Ah, nice catch. This actually explains a bug I was investigating where depending on the max streams limit, the client would still timeout sometimes. I moved the increments to after the checks (even for the remote streams case, even though in that case the connection would just fail anyway).